### DISCUSSION TITLE
Exposición «Desduplicación»: Preguntas y respuestas

### DISCUSSION THREAD

**Discussion by gwolf** [unamfi/sistop-2022-1#170] [Q&A]
Abro este foro como un espacio de discusión para la exposición que prepararon los alumnos Raúl Cruz y Miguel Ángel Villanueva, «Desduplicación». El video está disponible en:

https://youtu.be/WvWC6AuSRQE

Pueden consultar también el documento de texto en que los alumnos presentan este mismo tema:

https://github.com/unamfi/sistop-2022-1/blob/main/exposiciones/CruzRaul-VillanuevaMiguel/Desduplicacion.pdf

Además de participar aquí, les pido que llenen la pequña encuesta que está disponible en:

https://www.questionpro.com/t/ARiwuZqEWo

**Comment by CarlosGMendozaB** [comment]
Hola compañeros, me pareció muy interesante y buena su presentación. En la exposición nos hablan de le desduplicación a nivel de bloque, entonces, ¿Estos bloques deben tener una longitud especifica?.  Y al manejar estos bloques con una longitud en especifico, ¿Existe algún problema o ventaja al manejarlo así?

**Reply by Esparaul** [reply]
Nos alegra que te pareciera interesante la presentación :D. Respondiendo a tu pregunta, existen dos formas de desduplicación a nivel de bloque: la primera es con una longitud de bloque fija, la cual define bloques de tamaños fijo (normalmente van de los 4kB hasta los 64 kB), pero, la desventaja que se tiene al realizar esto, es que al aplicar los métodos hash en estos bloques fijos, disminuye considerablemente la probabilidad de encontrar bloques duplicados, esto debido a que no todos los bloques fijos contendrán exactamente la misma información. La segunda forma es seleccionando bloques de longitud variable, los cuales se dividen apoyándose de puntos clave en las secuencias de datos a analizar, de esta forma, encontraremos secuencias duplicadas con mayor efectividad, pero, debemos analizar el archivo completo para definir los límites de los bloques, por lo tanto, la desventaja que se tiene es que requeriremos mayor capacidad de procesamiento.
Espero que esto te sea de utilidad :).

**Comment by Inakyord** [comment]
Hola, espero estén bien. Me pareció muy interesante la presentación, sobre todo que se pueda usar diferente granularidad. Tendría dos preguntas: ¿ Además de servir para la desduplicación el hashing criptográfico utilizado nos da una medida de seguridad para nuestra información, o no realmente? Y la segunda, investigué que sí son posibles las colisiones con SHA-1 y MD5 a pesar de que es casi nula esta probabilidad. ¿Qué se hace para estos casos o se limita a perder la información por duplicidad errónea?

**Reply by MiguelVillanuev4** [reply]
Hola. La respuesta a la primer pregunta, es que sí; la intención de utilizar hash criptográfico, es que en caso de que se vean expuestas las cadenas devueltas por el hash, no sea posible desencriptar y regresar a la información original, aunque, ya se ha demostrado que algoritmos como SHA-1 ya se han logrado romper. Respondiendo a la segunda pregunta, en diversos documentos que consultamos, la respuesta tajante era que al presentarse la colisión, la información era descartada, provocando su pérdida.

**Reply by gwolf** [reply]
Me permito complementar un poco la respuesta. En primer lugar, MD5 ya no se usa. Este algoritmo duró muchos años, pero al paso del tiempo, además de que su tamaño se considera ya insficiente para aplicaciones criptográficas, se han encontrado vulnerabilidades que permiten realizar ataques de colisión de preimagen con muy poco esfuerzo.
SHA-1 se usa mucho hoy en día, aunque hay ya bases para suponer que va a ser roto relativamente pronto. Hay ya algoritmos sucesores a éste, como SHA-2 y SHA-3.
Y sí, la colisión de diferentes cadenas en un sólo punto tiene que ser vista como un evento catastrófico, pues resulta imposible protegerse contra ella. Afortunadamente, las matemáticas están de nuestro lado, y resulta casi imposible encontrar una colisión sobre un espacio de resultados tan grande como los hashes de la familia SHA.
